### ISSUE TITLE
prompt-toolkit 2.0 has changed some of their modules so not everything is where it once was

### ISSUE THREAD

**Issue by SolarAquarion** [xonsh/xonsh#2669]
https://github.com/jonathanslenders/python-prompt-toolkit/tree/master/prompt_toolkit/key_binding

**Comment by gforsyth** [comment]
Thanks @SolarAquarion !  We have a PR in progress at #2570 to address the new version.

Quick question -- did you upgrade `prompt_toolkit` or did version 2 get pulled down when you installed `xonsh`?  Because we tried to pin it to `<2` everywhere to avoid this but may have missed something.

**Comment by SolarAquarion** [comment]
I installed it myself

**Comment by asmodehn** [comment]
@gforsyth 
I did the install on windows, and installing xonsh did *not* retrieve prompt-toolkit (probably because it is optional ?)
After reading http://xon.sh/windows.html, I installed prompt-toolkit separately and got version 2 (default to latest).

Looking forward to #2570 and next release !

TL;DR : For those who are in a rush : `pip install prompt-toolkit<2`

**Comment by gforsyth** [comment]
Hey @asmodehn -- thanks for posting the workaround!
Yeah, since it's an optional dependency it won't get installed (using `pip`) unless you do `pip install xonsh[ptk]`
The conda-forge package includes it by default.  In any case, glad you got it working!

**Comment by gforsyth** [comment]
Whoops, sorry @SolarAquarion -- I just realized I never got back to you on this.  Sorry!

@asmodehn has the correct solution which is to install `prompt_toolkit<2` until we release a version compatible with ptk v2

**Comment by gforsyth** [comment]
Resolved in #2716
